llama_index/memory/redis_chat_memory_buffer.py

```diff
@@ -0,0 +1,322 @@
+import json
+import logging
+import sys
+from typing import Any, Callable, List, Optional
+from urllib.parse import urlparse
+
+from redis import Redis, RedisCluster
```

Comment:
This should be in a try/except import block, inside the functions where it is used

If you are using redis for type checking, you can keep this import too, but put it under a check

```
from typing import TYPE_CHECKING, ...

if TYPE_CHECKING:
    from redis import Redis, RedisCluster
```